iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -412,6 +412,7 @@ def createViewer(self, scriptUrl: str = DEFAULT_VIEWER_PATH, useStubViewer: bool
             xmlDocsByFilename = {
                 os.path.basename(self.outputFilename(doc.filepath)): deepcopy(doc.xmlDocument)
                 for doc in sorted(dts.modelDocument.referencesDocument.keys(), key=lambda x: x.objectIndex)
+                if doc.type == Type.INLINEXBRL
```

Comment:
> IXDS viewer currently includes all referenced docs, including schemas

If we don't include the iXBRL+linkbases, does this make the viewer no longer valid iXBRL?  I remember [this discussion](https://github.com/Arelle/ixbrl-viewer/pull/31#discussion_r287838780) from long ago.  We might have given up on that in a subsequent PR, but I've forgotten.

Perhaps it's separately useful to create a small architecture doc somewhere that documents goals and non-goals?  #379 and #386 also had some good discussions about our attempt maintain HTML+XHTML subset compatibility (non-goal of fixing input documents not written to that subset), and maintaining security posture of the input document (particularly for extracted text blocks: if input contains literal `<script>`, then extracted text blocks can too, but if input does not, then extracted text blocks must not).

Reply:
I think the problem that this is addressing a straightforward bug.  `iXBRLViewer.addfile()` is supposed to be for Inline XBRL files, and they'll get re-serialized using `XHTMLSerializer`, so it wasn't appropriate for schemas and linkbases to be included here.

For non-document set inputs, taxonomies are not included at all.

I'd be interested to know when this bug was introduced, as it certainly used to be possible to have a document set that didn't result in tabs for the taxonomy files.

I think it would be good to have an architecture document to address the points that you mention.  I also wonder whether we should reconsider the requirement for Inline XBRL validity of the output.

Reply:
I can't speak to that conversation or intent, but we already don't bundle local linkbases, only documents directly referenced by the iXBRL doc (we do provide a download for the original filing if the source is a zip, but that's separate).

@paulwarren-wk?

Reply:
> I'd be interested to know when this bug was introduced, as it certainly used to be possible to have a document set that didn't result in tabs for the taxonomy files.

I now see that this is a result of how Arelle is being invoked.

```
--file '[{"ixds":[{"file":"filing_documents.zip"}]}]'
```
If I'm understanding correctly, this is taking all the documents in the ZIP (including schemas and linkbases) and treating them as a document set, which seems a bit dubious, as only Inline XBRL documents can be part of a document set.  I wonder if this `ixds` notation should only look for HTML/XHTML in the ZIP, make a document set out of them, and then include the other files via DTS discovery?

That said, the fix on this PR still makes sense.